test/metabase/db/util_test.clj

```diff
@@ -0,0 +1,54 @@
+(ns metabase.db.util-test
+  (:require
+   [clojure.test :refer [deftest testing is]]
+   [metabase.db.util :as mdb.u]
+   [metabase.models.setting :refer [Setting]]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(deftest idempotent-insert!-test
```

Comment:
There's a chance `Setting` could be updated beyond `key` and `value` and in that world it would be helpful to understand why setting was chosen and why we're using key and value like this. 

It wouldn't hurt to be explicit about the assumptions we're making here too in a comment with the following info:
- Setting has two columns: `key` and `value`, which are both strings
- `key` has a uniqueness constraint on it, and `value` doesn't

Reply:
I didn't feel it necessary to call out that the fields are strings - would it have mattered if they were were numbers?

Reply:
Fair enough, I listed all the assumptions the test makes but probably not worth calling out to understand how it works at a high level

Reply:
Explaining the rationale was a good call out - let me know if you're happy with how I phrased it.

Reply:
Yes, looks good